File: chap3/Terminaison.js
// Ces programmes sont sous licence CeCILL-B V1.
var s,i,j,p;
s = 2;
p = false;
while (!p) {
	for (i = 1; i <= s - 1; i = i + 1) {
		j = s - i;
		if (i * i == 25 * j * j) {
			ISNprint(i);
			ISNprint("  ");
			ISNprintln(j);
			p = true;
		}
	}
	s = s + 1;
}

File: chap3/Terminaison.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let _ =
  let s = ref 2 in
  let p = ref false in
  while not !p do
    for i = 1 to  !s-1 do
      let j = !s - i in
      if i * i = 25 * j * j then begin
        print_int i; print_string " ";
        print_int j; print_newline ();
        p := true
      end;
    done;
    s := !s + 1
  done

